dashboard-server/queries/responses/response.test.js

```diff
@@ -57,34 +78,316 @@ describe('Response queries', () => {
+        timestamps[3]
+      }')`;
+
+      await vlabPool.query(MOCK_QUERY);
+
+      const mockData = {
+        email: user.email,
+        surveyName: survey.survey_name,
+        after: null,
+      };
+
+      const { email, surveyName, after } = mockData;
+
+      const res = await Response.all(email, surveyName, after);
+
+      res.responses.should.eql([
+        {
+          parent_surveyid: survey.id,
+          parent_shortcode: '101',
+          surveyid: survey.id,
+          flowid: '100001',
+          userid: '127',
+          question_ref: 'ref',
+          question_idx: '10',
+          question_text: 'text',
+          response: '{ "text": "last" }',
+          timestamp: timestamps[1],
+          metadata: null,
+          pageid: null,
+          translated_response: null,
+          token: token.encode([timestamps[1], '127', 'ref']),
+        },
+        {
+          parent_surveyid: survey.id,
+          parent_shortcode: '101',
+          surveyid: survey.id,
+          flowid: '100004',
+          userid: '127',
+          question_ref: 'ref',
+          question_idx: '10',
+          question_text: 'text',
+          response: '{ "text": "first" }',
+          timestamp: timestamps[2],
+          metadata: null,
+          pageid: null,
+          translated_response: null,
+          token: token.encode([timestamps[2], '127', 'ref']),
+        },
+        {
+          parent_surveyid: survey.id,
+          parent_shortcode: '101',
+          surveyid: survey.id,
+          flowid: '100004',
+          userid: '128',
+          question_ref: 'ref',
+          question_idx: '10',
+          question_text: 'text',
+          response: '{ "text": "first" }',
+          timestamp: timestamps[2],
+          metadata: null,
+          pageid: null,
+          translated_response: null,
+          token: token.encode([timestamps[2], '128', 'ref']),
+        },
+        {
+          parent_surveyid: survey.id,
+          parent_shortcode: '202',
+          surveyid: survey.id,
+          flowid: '100005',
+          userid: '126',
+          question_ref: 'ref',
+          question_idx: '10',
+          question_text: 'text',
+          response: '{ "text": "first" }',
+          timestamp: timestamps[3],
+          metadata: null,
+          pageid: null,
+          translated_response: null,
+          token: token.encode([timestamps[3], '126', 'ref']),
+        },
+      ]);
+
+      describe('surveyNotFound', () => {
+        it('should catch an error', async () => {
+          const surveyNotFound = 'this survey does not exist';
+          return expect(
+            Response.all(email, surveyNotFound, after),
+          ).to.be.rejectedWith(
+            `No responses were found for survey: ${surveyNotFound} for user: ${email}`,
+          );
+        });
+
+        it('should return a fulfilled promise', async () => {
+          return Promise.resolve(Response.all(email, surveyName, after)).should
```

Comment:
Promise.resolve? This looks very confusing. I'm not sure why something returns a resolved promise. It seems to me that Response.all returns a promise and you wait for it to be resolved to get the value...